subsys/shell/Kconfig.backends

```diff
@@ -0,0 +1,25 @@
+# Kconfig.backends - Shell badckends configuration options
+
+#
+# Copyright (c) 2018 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+menuconfig SHELL_BACKENDS
+	bool "Enable shell backends"
+	default y
+	help
+	  "Enable shell backends."
+
+if SHELL_BACKENDS
+
+config SHELL_BACKEND_SERIAL
+	bool "Enable serial backends."
+	default y
+	select SERIAL
```

Comment:
`depends on` is usually a better choice than `select` when visible symbols (symbols with prompts) are involved.

See https://github.com/zephyrproject-rtos/zephyr/pull/10212#discussion_r220336437 for discussion. It involves the same symbol.

Reply:
About this, please note that #10190 got reopened due to this:
https://github.com/zephyrproject-rtos/zephyr/issues/10127#issuecomment-425410780
https://github.com/zephyrproject-rtos/zephyr/issues/10127#issuecomment-425412420

Reply:
Ah, sorry, had missed the ping in https://github.com/zephyrproject-rtos/zephyr/issues/10190.

Just `default y if SHELL` on `SERIAL` would be problematic if `SHELL` needs `SERIAL`, as it would still be possible for the user to turn off `SERIAL` with `SHELL` enabled in that case.

I won't block this for `select`-related issues at least.